openmetadata-ui/src/main/resources/ui/playwright/e2e/Features/Permissions/DomainPermissions.spec.ts

```diff
@@ -38,67 +38,87 @@ const test = base.extend<{
 }>({
   page: async ({ browser }, use) => {
     const adminPage = await browser.newPage();
-    await adminUser.login(adminPage);
-    await use(adminPage);
-    await adminPage.close();
+    try {
+      await adminUser.login(adminPage);
+      await use(adminPage);
+    } finally {
+      await adminPage.close();
+    }
   },
   testUserPage: async ({ browser }, use) => {
     const page = await browser.newPage();
-    await testUser.login(page);
-    await use(page);
-    await page.close();
+    try {
+      await testUser.login(page);
+      await use(page);
+    } finally {
+      await page.close();
+    }
```

Comment:
why we are doing try catch here?

Reply:
Had added try catch here considering that if the login fails / test times out or any other error occurs, the page is always closed since Browser pages are expensive resources that consume memory and system resources.
Without the finally block, if an error occurs during testUser.login(page) or await use(page), the page would never be closed, leading to,
Memory leaks, Browser instances accumulating eventually running out of system resources.